src/components/include/sl.h

```diff
@@ -382,13 +412,22 @@ sl_cs_exit_schedule_nospin_arg(struct sl_thd *to)
 		if (unlikely(!pt)) t = sl__globals()->idle_thd;
 		else               t = sl_mod_thd_get(pt);
 	}
-	thdcap = t->thdcap;
-	prio   = t->prio;
 
+	if (t->type == SL_THD_AEP_TCAP || t->type == SL_THD_COMP_TCAP) {
```

Comment:
I agree. But I don't know of a better way to handle this and have no impact on the fast-path.